aws-xray-recorder-sdk-spring/src/main/java/com/amazonaws/xray/spring/aop/AbstractXRayInterceptor.java

```diff
@@ -0,0 +1,98 @@
+package com.amazonaws.xray.spring.aop;
+
+import com.amazonaws.xray.AWSXRay;
+import com.amazonaws.xray.entities.Segment;
+import com.amazonaws.xray.entities.Subsegment;
+import com.amazonaws.xray.exceptions.SegmentNotFoundException;
+import com.amazonaws.xray.strategy.ContextMissingStrategy;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Pointcut;
+
+import java.util.Optional;
+
+import static com.amazonaws.xray.AWSXRay.getCurrentSegmentOptional;
+
+public abstract class AbstractXRayInterceptor {
+
+    private static final Log logger = LogFactory.getLog(AbstractXRayInterceptor.class);
+
+    private static ContextMissingStrategy getContextMissingStrategy() {
+        return AWSXRay.getGlobalRecorder().getContextMissingStrategy();
+    }
+
+    private static Segment getCurrentSegment() {
+        Optional<Segment> segment = getCurrentSegmentOptional();
+        if (segment.isPresent()) {
+            return segment.get();
+        }
+        ContextMissingStrategy contextMissingStrategy = getContextMissingStrategy();
+        contextMissingStrategy.contextMissing("No segment in progress.", SegmentNotFoundException.class);
+        return null;
+    }
+
+    /**
+     * @param pjp the proceeding join point
+     * @return the result of the method being wrapped
+     * @throws Throwable
+     */
+    @Around("xrayTracedClasses() || xrayEnabledClasses()")
+    public Object traceAroundMethods(ProceedingJoinPoint pjp) throws Throwable {
+        return this.processXRayTrace(pjp);
+    }
+
+    protected Object processXRayTrace(ProceedingJoinPoint pjp) throws Throwable {
+        Segment segment = getCurrentSegment();
+        if (segment == null) {
+            AWSXRay.beginSegment(pjp.getSignature().getName());
```

Comment:
I noticed that this `beginSegment` remains while the `endSegment` logic was taken out. I'm still not sure we can get away with beginning the segments if one doesn't exist - I think we'd have to leave that up to the `SegmentContext` / `ContextMissingStrategy` chosen by the programmer.

Reply:
fixed and pushed